src/daft-io/src/local.rs

```diff
@@ -84,49 +98,83 @@ pub struct LocalFile {
 #[async_trait]
 impl ObjectSource for LocalSource {
     async fn get(&self, uri: &str, range: Option<Range<usize>>) -> super::Result<GetResult> {
-        const TO_STRIP: &str = "file://";
-        if let Some(p) = uri.strip_prefix(TO_STRIP) {
-            let path = std::path::Path::new(p);
-            Ok(GetResult::File(LocalFile {
-                path: path.to_path_buf(),
-                range,
-            }))
-        } else {
-            return Err(Error::InvalidFilePath {
-                path: uri.to_string(),
-            }
-            .into());
-        }
+        const LOCAL_PROTOCOL: &str = "file://";
+        let path = uri.strip_prefix(LOCAL_PROTOCOL).unwrap_or(uri);
+        Ok(GetResult::File(LocalFile {
+            path: path.into(),
+            range,
+        }))
     }
 
     async fn get_size(&self, uri: &str) -> super::Result<usize> {
-        const TO_STRIP: &str = "file://";
-        if let Some(p) = uri.strip_prefix(TO_STRIP) {
-            let path = std::path::Path::new(p);
-            let file = tokio::fs::File::open(path)
-                .await
-                .context(UnableToOpenFileSnafu {
-                    path: path.to_string_lossy(),
-                })?;
-            let metadata = file.metadata().await.context(UnableToOpenFileSnafu {
-                path: path.to_string_lossy(),
+        const LOCAL_PROTOCOL: &str = "file://";
+        let path = uri.strip_prefix(LOCAL_PROTOCOL).unwrap_or(uri);
+        let meta = tokio::fs::metadata(path)
```

Comment:
Changed to top-level `tokio::fs::metadata()` which (1) will follow symlinks, and (2) I believe has less overhead than opening + fetching the metadata off of the file object.